commit cdd1c1370d49a3e38ddebe0095249db72b2ecff7
Author: Your Name <jeffreyalaw@gmail.com>
Date:   Mon Nov 3 13:06:03 2025 -0700

    Revert "Build programs in $(others-noinstall) like tests"
    
    This reverts commit 14243c9db607652b53e08c894f647b8c1157aea0.

diff --git a/Rules b/Rules
index 7f6d063673..44c041c491 100644
--- a/Rules
+++ b/Rules
@@ -195,10 +195,9 @@ xtests:
 	  > $(objpfx)subdir-xtests.sum
 
 ifeq ($(build-programs),yes)
-binaries-all-notests = $(filter-out $(others-noinstall), \
-			 $(others) $(sysdep-others))
+binaries-all-notests = $(others) $(sysdep-others)
 binaries-all-tests = $(tests) $(tests-internal) $(xtests) $(test-srcs) \
-		     $(tests-container) $(others-noinstall)
+		     $(tests-container)
 binaries-all = $(binaries-all-notests) $(binaries-all-tests)
 binaries-static-notests = $(others-static)
 binaries-static-tests = $(tests-static) $(xtests-static)
@@ -386,7 +385,7 @@ $(objpfx)%.out: /dev/null $(objpfx)%	# Make it 2nd arg for canned sequence.
 # and pid namespaces) in which to run, should be added to
 # tests-container.
 $(tests-container:%=$(objpfx)%.out): $(objpfx)%.out : $(if $(wildcard $(objpfx)%.files),$(objpfx)%.files,/dev/null) $(objpfx)%
-	$(test-wrapper-env) $(run-program-env) $(test-via-rtld-prefix) \
+	$(test-wrapper-env) $(run-program-env) $(run-via-rtld-prefix) \
 	  $(common-objpfx)support/test-container env $(run-program-env) $($*-ENV) \
 	  $(host-test-program-cmd) $($*-ARGS) > $@; \
 	$(evaluate-test)
